client/src/components/Videos.js

```diff
@@ -0,0 +1,98 @@
+import { useState, useEffect } from "react";
+// import staticData from "../exampleresponse.json";
+import AddVideoForm from "./AddVideoForm";
+import Search from "./Search";
+import SingleVideo from "./SingleVideo";
+import "./Videos.css";
+
+const Videos = () => {
+  const [videosData, setVideosData] = useState([]);
+  const [searchValue, setSearchValue] = useState("");
+  const [showError, setShowError] = useState(false);
+
+  useEffect(() => {
+    fetch("http://localhost:8080/videos")
+      .then((res) => res.json())
+      .then((data) => setVideosData(data))
+      .catch((err) => console.error(err));
+  }, []);
+
+  // delete video (for SingleVideo)
+  const handleDeleteVideoClick = (id) => {
+    const newVideosData = videosData.filter((video) => video.id !== id);
+    setVideosData(newVideosData);
+  };
+
+  const handleSearch = (event) => {
+    setSearchValue(event.target.value);
+  };
+
+  // matchYoutubeUrl was copied from here: https://www.codegrepper.com/code-examples/javascript/javascript+validate+url+to+match+youtube+video
+  const matchYoutubeUrl = (url) => {
+    var p =
+      /^(?:https?:\/\/)?(?:m\.|www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
+    if (url.match(p)) {
+      return url.match(p)[1];
+    }
+    return false;
+  };
+
+  const handleAddVideo = async (title, url) => {
+    if (title !== "" && url !== "" && matchYoutubeUrl(url)) {
```

Comment:
This works, but often times we'll write this like:
```js
if (title === "" || url === "" || !matchYoutubeUrl(url)) {
  setShowError(true);
  return;
}
try {
  ...
}
```

Can you talk through why that may be more clear?